controllers/bsl.go

```diff
@@ -196,9 +197,76 @@ func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool,
 			)
 		}
 	}
+
+	_, err := r.syncBSLs(&dpa)
```

Comment:
I studied the difference between BSLs created by DPA and by users, and I think only difference is these metadata that DPA includes to the objects
```yaml
metadata:
  labels:
    app.kubernetes.io/component: bsl
    app.kubernetes.io/instance: <bsl-name>
    app.kubernetes.io/managed-by: oadp-operator
    app.kubernetes.io/name: oadp-operator-velero
    openshift.io/oadp: "True"
    openshift.io/oadp-registry: "True"
  ownerReferences:
  - apiVersion: oadp.openshift.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: DataProtectionApplication
    name: <dpa-name>
    uid: <uid>
``` 

so looking labels is a good idea (BUT ALSO NEED TO CHECK NAMESPACE, not?)

Did not understand what the current solution is doing, I suggest simplifying

What we need is to do is:
- store the BSLs names created by DPA , in OADP namespace
- list BSLs created by DPA, in OADP namespace
- if count differs, delete BSLs not defined in DPA

right?

Something like this would solve the problem?
```go
	dpaBSLNames := []string{}
	// Loop through all configured BSLs
	for i, bslSpec := range dpa.Spec.BackupLocations {
		// Create BSL as is, we can safely assume they are valid from
		// ValidateBackupStorageLocations

		// check if BSL name is specified in DPA spec
		bslName := fmt.Sprintf("%s-%d", r.NamespacedName.Name, i+1)
		if bslSpec.Name != "" {
			bslName = bslSpec.Name
		}
		dpaBSLNames = append(dpaBSLNames, bslName)

		...
	}

	dpaBSLs := velerov1.BackupStorageLocationList{}
	dpaBSLLabels := map[string]string{
		"app.kubernetes.io/name":       common.OADPOperatorVelero,
		"app.kubernetes.io/managed-by": common.OADPOperator,
		"app.kubernetes.io/component":  "bsl",
	}
	err := r.List(r.Context, &dpaBSLs, client.InNamespace(r.NamespacedName.Namespace), client.MatchingLabels(dpaBSLLabels))
	if err != nil {
		return false, err
	}
	if len(dpaBSLNames) != len(dpaBSLs.Items) {
		for _, bsl := range dpaBSLs.Items {
			if !slices.Contains(dpaBSLNames, bsl.Name) {
				if err := r.Delete(r.Context, &bsl); err != nil {
					return false, err
				}
				// event recorder
			}
		}
	}

	return true, nil
}
```

Reply:
@mateusoliveira43 This solution should also work, basically it's the same thing,

You are getting the current BSLs in env. using the labels, and then looping over Dpa spec BSLs, and matching the names with slice variable stored.

I am also getting the current BSLs in env. using the labels, just that I am looping the other way around on the BSLs present in the environment, and then checking the name format using the same (length of dpaBSL.Items)

It really looks like the other way of doing the same thing to me..
Do you want me to change and test again?

Reply:
But yes, it is definitely easier to understand, and lighter. So....changes pushed. 

Thanks @mateusoliveira43 !!